magenta-lib/src/main/scala/magenta/tasks/ASGTasks.scala

```diff
@@ -123,7 +123,25 @@ case class CullInstancesWithTerminationTag(info: AutoScalingGroupInfo, region: R
   override def execute(asg: AutoScalingGroup, resources: DeploymentResources, stopFlag: => Boolean, asgClient: AutoScalingClient) {
     EC2.withEc2Client(keyRing, region, resources) { ec2Client =>
       ELB.withClient(keyRing, region, resources) { elbClient =>
-        val instancesToKill = asg.instances.asScala.filter(instance => EC2.hasTag(instance, "Magenta", "Terminate", ec2Client))
+        val instancesToKill = asg.instances.asScala
+          .filter(instance => {
+            if(instance.lifecycleState == LifecycleState.UNKNOWN_TO_SDK_VERSION) {
+              logger.warn(s"Instance lifecycle state ${instance.lifecycleStateAsString} isn't recognised in the AWS SDK. Is there a later version of the AWS SDK available?")
+            }
```

Comment:
Taking a similar approach as https://github.com/guardian/prism/pull/162 to avoid possible NPEs.